### PR TITLE
Print count of failed tests and list them at the exit of test suite

### PR DESCRIPTION
**Pull request by sebastianbaginski**
For readability of test results.

### CODE REVIEW

**Comment by sebastianbaginski** [comment]
I hope AppVeyor fails on this one :)

**Comment by sebastianbaginski** [comment]
Thanks appveyor, nice to see it works :)

**Comment by marcusnaslund** [comment]
If you want to try, you now have access to colouring using ASCII escape codes in `source/io/Terminal`

**Comment by marcusnaslund** [comment]
Validated @davidhesselbom

File: source/unit/Fixture.ooc [hunk 1]
```diff
@@ -21,9 +21,25 @@ import Constraints
...
 Fixture: abstract class {
 	totalTime: static Double
+	exitHandlerRegistered: static Bool
+	failureNames: static VectorList<String>
 	name: String
 	tests := VectorList<Test> new()
-	init: func (=name)
+	printFailures: static func {
+		if (This failureNames && (This failureNames count > 0)) {
+			"Failed tests: %i [" format(This failureNames count) print()
```
Comment (marcusnaslund) [inline source/unit/Fixture.ooc]: `printfln(This failureNames count)` would do the same thing, FYI

File: source/unit/Fixture.ooc [hunk 2]
```diff
@@ -21,9 +21,25 @@ import Constraints
...
+			for (i in 0 .. This failureNames count - 1)
+				(This failureNames[i] + ", ") print()
+			This failureNames[This failureNames count - 1] print()
+			"]" println()
+		}
+	}
+	init: func (=name) {
+		if (!This exitHandlerRegistered) {
+			This exitHandlerRegistered = true
+			atexit(This printFailures)
```
Comment (davidhesselbom) [inline source/unit/Fixture.ooc]: This one's new to me. Interesting!

File: source/unit/Fixture.ooc [hunk 3]
```diff
@@ -21,9 +21,25 @@ import Constraints
...
 	name: String
 	tests := VectorList<Test> new()
-	init: func (=name)
+	printFailures: static func {
+		if (This failureNames && (This failureNames count > 0)) {
+			"Failed tests: %i [" format(This failureNames count) print()
+			for (i in 0 .. This failureNames count - 1)
+				(This failureNames[i] + ", ") print()
+			This failureNames[This failureNames count - 1] print()
+			"]" println()
```
Comment (davidhesselbom) [inline source/unit/Fixture.ooc]: `This failureNames count -1 + ']') println()`?

Comment (davidhesselbom) [inline source/unit/Fixture.ooc]: Or, to maintain the style on line 30,
```ooc
"%s]" format(This failureNames[This failureNames count - 1]) println()
```

Comment (davidhesselbom) [inline source/unit/Fixture.ooc]: But, if you don't want to, leave it as is.

Comment (marcusnaslund) [inline source/unit/Fixture.ooc]: If we're changing this anyhow, I do prefer `printfln(...)` over `format(...) println()`

Comment (davidhesselbom) [inline source/unit/Fixture.ooc]: Yeah, me too. Goes for both line 30 and line 34.

Comment (sebastianbaginski) [inline source/unit/Fixture.ooc]: OK

**Comment by sebastianbaginski** [comment]
updated

**Pull request merged**